backend/localization.py

```diff
@@ -1,17 +1,43 @@
 import json
 import os
 
-BASE_PATH = os.path.join(os.path.dirname(__file__), "locales")
-DEFAULT_LANG = "en"
-
-def load_translation() -> dict:
-    lang_code="en"
-    lang_path = os.path.join(BASE_PATH, lang_code, "strings.json")
-    fallback_path = os.path.join(BASE_PATH, DEFAULT_LANG, "strings.json")
-
-    try:
-        with open(lang_path, 'r', encoding='utf-8') as f:
-            return json.load(f)
-    except FileNotFoundError:
-        with open(fallback_path, 'r', encoding='utf-8') as f:
-            return json.load(f)
+BASE_PATH = os.path.join(os.path.dirname(__file__), 'locales')
+DEFAULT_LANG = 'en'
+
+def load_translation(lang_code: str = DEFAULT_LANG, reload: bool = False) -> dict[str, str]:
+    """
+    Load and cache translation JSON for a given language.
+
+    Args:
+        lang_code: 2-letter language code (e.g. "en", "ua").
+        reload:    If True, forces re-reading from disk even if cached.
+
+    Returns:
+        A dict with translation strings.
+
+    Raises:
+        FileNotFoundError: if neither lang_code nor DEFAULT_LANG file exists.
+    """
+    # init cache dict on function
+    cache = getattr(load_translation, '_cache', {})
+
+    # if not reloading and we already have it, return cached
+    if not reload and lang_code in cache:
+        return cache[lang_code]
+
+    # build candidate file paths: preferred then fallback
+    candidates = [
+        os.path.join(BASE_PATH, lang_code, 'strings.json'),
+        os.path.join(BASE_PATH, DEFAULT_LANG, 'strings.json'),
+    ]
+
+    for path in candidates:
+        if os.path.isfile(path):
+            with open(path, 'r', encoding='utf-8') as f:
+                data = json.load(f)
+            cache[lang_code] = data
+            load_translation._cache = cache
+            return data
```

Comment:
This function just returns a dict which is directly accessed everywhere it is used. If we attempt to load a language that is only partially translated, the dict will be missing keys, and accessing them will throw and error and crash the program. Given that the default english translation should not be missing any keys, I think it would be best to change this to load english first, and then apply the desired language over top of that. That should both prevent crashes and present a string to the user instead of nothing, even if they can't understand it.

Additionally a warning should probably be logged if we try and load a language but cannot find a file for it.